File: backend/src/main/java/cha_revelacao/dto/response/EventoResponse.java
package cha_revelacao.dto.response;

import cha_revelacao.model.Evento;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class EventoResponse {
    private Long id;
    private String titulo;
    private String descricao;
    private LocalDateTime dataEvento;
    private String local;
    private String nomeMae;
    private String nomePai;
    private Boolean revelado;
    private Evento.SexoBebe resultadoRevelacao;
    private Evento.StatusEvento status;
    private Boolean votacaoEncerrada;
    private LocalDateTime dataEncerramentoVotacao;
    private LocalDateTime criadoEm;
    private UsuarioResponse organizador;
    private Integer totalVotosMenino;
    private Integer totalVotosMenina;
    private Integer totalVotos;

    public UsuarioResponse getOrganizador() { return organizador; }
    public void setOrganizador(UsuarioResponse organizador) { this.organizador = organizador; }
    public Integer getTotalVotosMenino() { return totalVotosMenino; }
    public void setTotalVotosMenino(Integer totalVotosMenino) { this.totalVotosMenino = totalVotosMenino; }
    public Integer getTotalVotosMenina() { return totalVotosMenina; }
    public void setTotalVotosMenina(Integer totalVotosMenina) { this.totalVotosMenina = totalVotosMenina; }
    public Integer getTotalVotos() { return totalVotos; }
    public void setTotalVotos(Integer totalVotos) { this.totalVotos = totalVotos; }
}

File: backend/src/DTOs/Responses/EventoResponse.cs
namespace MemuVie.Evento.DTOs.Responses;

using MemuVie.Evento.Models;

public class EventoResponse
{
    public long Id { get; set; }
    public string Titulo { get; set; } = string.Empty;
    public string? Descricao { get; set; }
    public DateTime DataEvento { get; set; }
    public string? Local { get; set; }
    public string NomeMae { get; set; } = string.Empty;
    public string NomePai { get; set; } = string.Empty;
    public bool Revelado { get; set; }
    public BabySex? ResultadoRevelacao { get; set; }
    public EventStatus Status { get; set; }
    public bool VotacaoEncerrada { get; set; }
    public DateTime? DataEncerramentoVotacao { get; set; }
    public string? FotoCapa { get; set; }
    public string? VideoDestaque { get; set; }
    public string? CorTema { get; set; }
    public EventType TipoEvento { get; set; }
    public DateTime CriadoEm { get; set; }
    public UsuarioResponse Usuario { get; set; } = null!;
}
